### PR TITLE
tools: Add funcinterval bcc tool.

### PR DESCRIPTION
**Pull request by netedwardwu**
Time interval between the same function as a histogram.
Referenced from funclatency.

### CODE REVIEW

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Comment by yonghong-song** [comment]
[buildbot, ok to test]

File: tools/funcinterval.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+# signal handler
+def signal_ignore(signal, frame):
+    print()
+
+
+# load BPF program
+b = BPF(text=bpf_text)
+
+# attach probes
+b.attach_kprobe(event_re=func_name, fn_name="trace_func_entry")
```
Comment (yonghong-song) [inline tools/funcinterval.py:125]: Here, we are matching as regex. Maybe you have an option to control this? Sometimes people really just want to track a particular function.

**Comment by yonghong-song** [comment]
cc @brendangregg

**Comment by brendangregg** [comment]
Well this is easy: I already know it'll be a useful tool as I'm using something similar (bpftrace one-liner) to check arrival distributions. Plus the _example.txt is good. I'm happy with this, thanks.

Like funccount(8), we'll want to do tracepoints as well. That functionality can be added in a later PR.

**Comment by netedwardwu** [comment]
Hi @yonghong-song ,

As you noticed, I remove some funclatency features on funcinterval. Because of those features I never used on funclatency, even I solved many issues by funclatency.
I used funclatency only when I understand the purpose of the specific function name.

I prefer leaving those features to other developers to contribute.
BUT, if you insist on it. For sure that I can do it.
Thanks for your code review.

Hi @brendangregg ,

Thanks for your comment. Yes, the tracepoint feature is what I wanted to do.
I will do another PR for it.

**Comment by netedwardwu** [comment]
Hi all,

I have added tracepoint feature, please help to check it.

Thanks

**Comment by yonghong-song** [comment]
Looks good. Thanks!

**Pull request merged**